### PR TITLE
feat: aws integration

### PR DESCRIPTION
**Pull request by AlexsJones**
Closes # 

## ✅ Checks

- [ x] My pull request adheres to the code style of this project
- [x ] My code requires changes to the documentation
- [x ] I have updated the documentation as required
- [x ] All the tests have passed

## ℹ Additional Information

This initial integration will surface issues found in EKS beneath the current cluster

### CODE REVIEW

File: pkg/integration/aws/aws.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		session: a.sess,
+	}
+}
+
+func (a *AWS) GetAnalyzerName() []string {
+
+	return []string{"EKS"}
+}
+
+func (a *AWS) GetNamespace() (string, error) {
```
Comment (JuHyung-Son) [inline pkg/integration/aws/aws.go:47]: what is this namespace?
k8s namespace or something?

File: pkg/integration/aws/eks.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+type EKSAnalyzer struct {
+	session *session.Session
+}
+
+func (e *EKSAnalyzer) Analyze(analysis common.Analyzer) ([]common.Result, error) {
+	var cr []common.Result = []common.Result{}
+	_ = map[string]common.PreAnalysis{}
+	svc := eks.New(e.session)
+	// Get the name of the current cluster
+	kubecConfigPath := filepath.Join(os.Getenv("HOME"), ".kube", "config")
```
Comment (JuHyung-Son) [inline pkg/integration/aws/eks.go:24]: typo: kubecConfigPath

**Review by JuHyung-Son — COMMENTED** [review]
awesome.

i have a question and a commnet.

what can we analyze only for eks? for cluster analyze, it is just same as normal k8sgpt since eks is cluster.
maybe we can use aws eks package itself not just k8s client.

and why not using k8sgpt global config `kubeconfg`? and fix .kube/config

File: pkg/integration/aws/aws.go (hunk 1 above)
Comment (AlexsJones) [reply pkg/integration/aws/aws.go:47]: This is redundant for this type of integration. 
The original idea was integrations would be deployed into a K8s namespace.

File: pkg/integration/aws/aws.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+func (a *AWS) UnDeploy(namespace string) error {
+	a.sess = nil
+	return nil
+}
+
+func (a *AWS) AddAnalyzer(mergedMap *map[string]common.IAnalyzer) {
+	// Check for AWS credentials in the environment
+	// https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html
+	if os.Getenv("AWS_ACCESS_KEY_ID") == "" || os.Getenv("AWS_SECRET_ACCESS_KEY") == "" {
```
[the remark this answers is not in the document]
Comment (AlexsJones) [reply pkg/integration/aws/aws.go:28]: That's true.. perhaps we should allow `AWS_PROFILE` as a variable

File: pkg/integration/aws/eks.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+type EKSAnalyzer struct {
+	session *session.Session
+}
+
+func (e *EKSAnalyzer) Analyze(analysis common.Analyzer) ([]common.Result, error) {
+	var cr []common.Result = []common.Result{}
+	_ = map[string]common.PreAnalysis{}
+	svc := eks.New(e.session)
+	// Get the name of the current cluster
+	kubeConfigPath := filepath.Join(os.Getenv("HOME"), ".kube", "config")
```
[the remark this answers is not in the document]
Comment (AlexsJones) [reply pkg/integration/aws/eks.go:24]: How would you access that in this context?

[the remark this answers is not in the document]
Comment (JuHyung-Son) [reply pkg/integration/aws/eks.go:24]: hmm if we can not access config.. how about make it additional argument.
i think kubeconfig should be same with the one from cli. `k8sgpt --kubeconfig=$HOME/.kube/other_config`

[the remark this answers is not in the document]
Comment (AlexsJones) [reply pkg/integration/aws/eks.go]: What do you think about this? 
```
	var kubeconfig string
	kubeconfigFromPath := viper.GetString("kubeconfig")
	if kubeconfigFromPath != "" {
		kubeconfig = kubeconfigFromPath
	} else {
		kubeconfig = filepath.Join(os.Getenv("HOME"), ".kube", "config")
	}
```

[the remark this answers is not in the document]
Comment (JuHyung-Son) [reply pkg/integration/aws/eks.go]: good!

File: pkg/integration/aws/aws.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (JuHyung-Son) [reply pkg/integration/aws/aws.go:28]: does `AWS_PROFILE` will be added this pr? or next?

**Pull request merged**